bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/discovery/SensorDiscoveryService.java

```diff
@@ -0,0 +1,139 @@
+            ((EcobeeThermostatBridgeHandler) handler).setDiscoveryService(this);
+            bridgeHandler = (EcobeeThermostatBridgeHandler) handler;
+        }
+    }
+
+    @Override
+    public @Nullable ThingHandler getThingHandler() {
+        return bridgeHandler;
+    }
+
+    @Override
+    public void activate() {
+        logger.debug("SensorDiscovery: Activating Ecobee sensor discovery service");
+    }
+
+    @Override
+    public void deactivate() {
+        logger.debug("SensorDiscovery: Deactivating Ecobee sensor discovery service");
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_SENSOR_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public void startBackgroundDiscovery() {
+        EcobeeThermostatBridgeHandler localBridgeHandler = bridgeHandler;
+        if (localBridgeHandler == null) {
+            logger.info("SensorDiscovery: Can't perform background discovery because bridgeHandler is null");
+            return;
+        }
+        logger.debug("SensorDiscovery: Performing background discovery scan for {}",
+                localBridgeHandler.getThing().getUID());
+        discoverSensors();
+    }
+
+    @Override
+    public void startScan() {
+        EcobeeThermostatBridgeHandler localBridgeHandler = bridgeHandler;
+        if (localBridgeHandler == null) {
+            logger.info("SensorDiscovery: Can't perform discovery scan because bridgeHandler is null");
+            return;
+        }
+        logger.debug("SensorDiscovery: Starting discovery scan for {}", localBridgeHandler.getThing().getUID());
+        discoverSensors();
+    }
+
+    @Override
+    public synchronized void abortScan() {
+        super.abortScan();
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+    }
+
+    private String buildLabel(String name) {
+        return String.format("Ecobee Sensor %s", name);
+    }
+
+    private synchronized void discoverSensors() {
+        EcobeeThermostatBridgeHandler localBridgeHandler = bridgeHandler;
+        if (localBridgeHandler == null) {
+            logger.info("SensorDiscovery: Can't discover sensors because bridgeHandler is null");
+            return;
+        }
+        for (RemoteSensorDTO sensor : localBridgeHandler.getSensors()) {
+            ThingUID bridgeUID = localBridgeHandler.getThing().getUID();
+            ThingUID sensorUID = new ThingUID(UID_SENSOR_THING, bridgeUID, sensor.id.replace(":", "-"));
+            thingDiscovered(createDiscoveryResult(sensorUID, bridgeUID, sensor));
+            logger.trace("SensorDiscovery: Sensor with id '{}' and name '{}' added to Inbox with UID '{}'", sensor.id,
+                    sensor.name, sensorUID);
+        }
+    }
+
+    private DiscoveryResult createDiscoveryResult(ThingUID sensorUID, ThingUID bridgeUID, RemoteSensorDTO sensor) {
+        Map<String, Object> properties = new HashMap<>(0);
```

Comment:
I'm not sure what good setting to 0 would do, but if you want to prevent the map from resizing then setting it to double the expected size works. The make will resize the cache once the fill exceeds 75% so by making the size double what you expect you make sure it won't exceed 50% fill, thus preventing rehashing.
```suggestion
        Map<String, Object> properties = new HashMap<>(2);
```

Reply:
It was copy/pasted from somewhere else. Fixed.